$MirOS$
$NetBSD: patch-aa,v 1.29 2008/06/22 23:38:18 dholland Exp $

	Some people never learn that stuff linking in C++ libraries
	must be linked with the C++ linker

--- Make.rules.in.orig	Tue Jun 17 14:07:08 2008
+++ Make.rules.in	Tue Jul 29 21:43:51 2008
@@ -72,7 +72,11 @@ MAKECTESTS   = $(TOOLSDIR)/tools/make_ct
 WRC          = $(TOOLSDIR)/tools/wrc/wrc
 WMC          = $(TOOLSDIR)/tools/wmc/wmc
 WIDL         = $(TOOLSDIR)/tools/widl/widl
+ifeq (${CXXLINK},1)
+WINEGCC      = $(TOOLSDIR)/tools/winegcc/wineg++
+else
 WINEGCC      = $(TOOLSDIR)/tools/winegcc/winegcc
+endif
 RELPATH      = $(TOOLSDIR)/tools/relpath
 SFNT2FNT     = $(TOOLSDIR)/tools/sfnt2fnt
 FNT2FON      = $(TOOLSDIR)/tools/fnt2fon
@@ -120,7 +124,7 @@ IDL_GEN_HEADERS = $(IDL_H_SRCS:.idl=.h) 
 CLEAN_TARGETS = $(IDL_GEN_C_SRCS) $(IDL_GEN_HEADERS) $(IDL_TLB_SRCS:.idl=.tlb) $(IDL_P_SRCS:%=dlldata.c) \
                 $(BISON_SRCS:.y=.tab.c) $(BISON_SRCS:.y=.tab.h) $(LEX_SRCS:.l=.yy.c) $(MC_SRCS:.mc=.mc.rc)
 
-OBJS = $(C_SRCS:.c=.o) $(BISON_SRCS:.y=.tab.o) $(LEX_SRCS:.l=.yy.o) $(IDL_GEN_C_SRCS:.c=.o) $(EXTRA_OBJS)
+OBJS = $(EXTRA_OBJS) $(C_SRCS:.c=.o) $(BISON_SRCS:.y=.tab.o) $(LEX_SRCS:.l=.yy.o) $(IDL_GEN_C_SRCS:.c=.o)
 
 RCOBJS = $(RC_SRCS:.rc=.res.o)
 LINTS  = $(C_SRCS:.c=.ln)
